monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -821,6 +838,23 @@ object Observable {
   def concatDelayError[A](sources: Observable[A]*): Observable[A] =
     Observable.fromIterable(sources).concatMapDelayError[A](identity)
 
+  /**
+    * like `zipWith2`, but when `oa1` completes before `oa2` does, the new observable will
+    * continue emitting items as `f(paddingA1, oa2.onNext())` until `oa2` completes;
+    * and vice versa, when `oa2` completes before `oa1` does, the new observable will
+    * continue emitting items as `f(oa1.onNext(), paddingA2)` until `oa1` completes.
+    */
+  def pad2With[A1,A2,R](oa1: Observable[A1], oa2: Observable[A2], paddingA1: ⇒ A1, paddingA2: ⇒ A2)(f: (A1,A2) ⇒ R): Observable[R] =
+    new builders.Pad2Observable(oa1, oa2, paddingA1, paddingA2)(f)
```

Comment:
Interesting. Two changes:

1. the number that indicates the arity, in this case "2", needs to be last, because we've used this as a convention. In other words I'd prefer `padWith2`. Because it is `zip2` and `zipWith2` and `combineLatest2` and `combineLatestWith2`. The reason is that, normally such functions are overloads and so other libraries simply make them multi-variadic, but I don't particularly like Java-style overloads and the number comes as a suffix to make it easier to change (e.g. you add a new param and you change 2 to 3), since the number being the suffix, you can move the keyboard cursor fast enough - and I've seen other libraries do this as well - stupid reason, but at least there's one :-)
2. would be better to make those by-name params `Coeval` - it's a superficial change, but by-name params IMHO are error-prone and by using [Coeval](https://monix.io/docs/2x/eval/coeval.html) we can control how evaluation happens

```scala
// Original:
pad2With(oa1, oa2, someA, someB)

// Modified:
padWith2(oa1, oa2, Coeval(someA), Coeval(someB))
```

Everything is subject for discussion btw and this type is new and so its usage is inconsistent for now.